app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,55 @@
+class CommentsController < ApplicationController
+  before_action :set_comment, only: %i[edit update destroy]
+  before_action :set_task, only: %i[create destroy]
+  before_action :set_project
+  before_action -> { authorize! @project, with: TaskPolicy }
+
+  def create
+    authorize! Task.find(comment_params[:task_id]).project, with: TaskPolicy
+    @comment = Comment.new(comment_params)
+
+    if @comment.save
+      redirect_to project_task_url(@task.project, @task), notice: "Comment was successfully created!"
+    else
+      flash.now[:notice] = "Something went wrong. Try again."
+      render "tasks/show"
+    end
+  end
+
+  def destroy
+    @comment.destroy
+    redirect_to project_task_url(@project, @task), notice: "Comment was successfully destroyed."
+  end
+
+  def edit
+  end
+
+  def update
+    authorize! Task.find(comment_params[:task_id]).project, with: TaskPolicy
+    if @comment.update(comment_params)
+      redirect_to project_task_url(@comment.task.project, @comment.task),
```

Comment:
You use `Task.find(comment_params[:task_id]).project` and `@comment.task.project`. It is the same project, right? Why you can use just @project?